internal/cmd/stack/open_command.go

```diff
@@ -0,0 +1,234 @@
+package stack
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/errors"
+	"github.com/shurcooL/graphql"
+	"github.com/urfave/cli/v2"
+
+	"github.com/spacelift-io/spacectl/client/structs"
+	"github.com/spacelift-io/spacectl/internal"
+	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
+)
+
+func openCommandInBrowser(cliCtx *cli.Context) error {
+	ignoreSubdir := cliCtx.Bool(flagIgnoreSubdir.Name)
+	getCurrentBranch := cliCtx.Bool(flagCurrentBranch.Name)
+	count := cliCtx.Int(flagSearchCount.Name)
+
+	var subdir *string
+	if !ignoreSubdir {
+		got, err := getGitRepositorySubdir()
+		if err != nil {
+			return err
+		}
+		subdir = &got
+	}
+
+	var branch *string
+	if getCurrentBranch {
+		got, err := getGitCurrentBranch()
+		if err != nil {
+			return err
+		}
+		branch = &got
+	}
+
+	name, err := getRepositoryName()
+	if err != nil {
+		return err
+	}
+
+	return findAndSelect(cliCtx.Context, &stackSearchParams{
+		count:          count,
+		projectRoot:    subdir,
```

Comment:
Does this correctly interact with empty project roots?

Reply:
I have stacks: 
![image](https://user-images.githubusercontent.com/40318863/219005284-011c2229-2b46-4057-91c4-cd67dce839a7.png)

If I execute open in dir of a repo I get:
<img width="457" alt="image" src="https://user-images.githubusercontent.com/40318863/219005353-b59b2bbb-953e-4ceb-86cc-5b8247261d2e.png">